Scripts/RemediationScripts/Remediate-EnableHttpsForStorageAccount.ps1

```diff
@@ -0,0 +1,723 @@
+      
+        .EXAMPLE
+        Enable-StorageEncryptionInTransit -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 
+      
+        .EXAMPLE
+        PS> Enable-HttpsForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000  -FilePathC:\Users\Documents\AzTS\Remediation\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\20211013_0608\EnableSecureTransit\StorageWithDisableHTTPs.csv
+         
+        .LINK
+        None
+    #>
+
+    param (
+        [string]
+        [Parameter(Mandatory = $true, HelpMessage="Enter subscription id for remediation")]
+        $SubscriptionId,
+
+        [string]
+        [Parameter(Mandatory = $false, HelpMessage="Enter path for the CSV file")]
+        $FilePath,
+   
+        [string]
+		[Parameter(Mandatory = $false, HelpMessage="Comma separated resource group name(s) to be excluded from remediation")]
+		$ExcludeResourceGroupNames,
+
+		[string]
+		[Parameter(Mandatory = $false, HelpMessage="Comma separated resource name(s) to be excluded from remediation")]
+		$ExcludeResourceNames,
+        
+        [switch]
+        [Parameter(Mandatory = $false, HelpMessage="Specifies a forceful remediation without any prompts")]
+        $Force,
+
+        [switch]
+        [Parameter(Mandatory = $false)]
+        $DryRun
+    )
+
+    write-host " $([Constants]::DoubleDashLine)"
+
+    try
+    {   
+        Write-Host "[Step 1 of 6]: Checking for pre-requisites..." 
+        Setup-Prerequisites
+        Write-Host $([Constants]::SingleDashLine)  
+    }   
+    catch 
+    {
+        Write-Host "Error occurred while checking pre-requisites. ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)    
+        break
+    }
+    
+  
+    #Connect to AzAccount
+    $isContextSet = Get-AzContext
+    if ([string]::IsNullOrEmpty($isContextSet))
+    {  
+        Write-Host $([Constants]::SingleDashLine)    
+        Write-Host "Connecting to AzAccount..."
+        Connect-AzAccount -ErrorAction Stop
+        Write-Host "Connected to AzAccount" -ForegroundColor $([Constants]::MessageType.Update)
+    }
+
+    # Setting context for current subscription.
+    $currentSub = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop -Force
+  
+    Write-Host "Metadata Details: `nSubscriptionName: $($currentSub.Subscription.Name) `nSubscriptionId: $($SubscriptionId) `nAccountName: $($currentSub.Account.Id) `nAccountType: $($currentSub.Account.Type)" 
+  
+    Write-Host $([Constants]::SingleDashLine)  
+    Write-Host "Starting with subscription [$($SubscriptionId)]..." -ForegroundColor $([Constants]::MessageType.Update)
+    
+    Write-Host $([Constants]::DoubleDashLine)
+
+    write-host "[Step 2 of 6] : Validating the account type."
+   
+    Write-Host "*** To enable HTTPS for Storage Account in a Subscription, Contributor and higher privileges on the Storage Account are required. ***" -ForegroundColor $([Constants]::MessageType.Info)
```

Comment:
Same as above - "To enable secure transfer" (or) "To encrypt in transit...".

Reply:
Done